src/pages/_app.tsx

```diff
@@ -1,9 +1,19 @@
 import type { AppProps } from 'next/app'
+import { useRouter } from 'next/router'
+import { Layout } from '@/components/layout/Layout'
 import '@/styles.scss'
 
+const errorPagesUrls = '/_error'
```

Comment:
What's the purpose of this constant?

Reply:
The purpose of this constant is to not display the layout component (header and footer) for error pages (404 and 500 pages).

Reply:
OK, but is `/_error` a next convention?
Is this route declared somewhere?

Reply:
Yes, to detect the errors page from `app.tsx` in next js the router pathname for errors page is `/_error`

Reply:
But, as far as I can see, there is no component related to the ` /_error` path no?
In the nextjs [documentation](https://nextjs.org/docs/pages/building-your-application/routing/custom-error), this path refers to the component in the `page` folder, here what do you render ?

Reply:
And I will add that it is an UX question to remove (or not) the header and footer components when an error occurs.

Because, by doing this, there is no more active navigation links in the app. The most part of recent web apps keep header and footer elements and only render a specific page in the main content to handle errors.

@SpicySalm0n @MdechampG
Please, what's your point of view about this subject?

Reply:
The 404 and 500 pages are generated by default by next js, so we can't pass the `urls` for footer to these pages. 
So in order to pass the header and footer to the error pages, we must create a custom page for 404 and 500 as mentioned here in the [documentaion](https://nextjs.org/docs/pages/building-your-application/routing/custom-error) and then pass the error in the main content.

Reply:
There is another point we can also also handle the error for 404 and 500 page by getting the status code for error and  display only the component without checking the pathname.

```tsfunction MyApp({ Component, pageProps }: AppProps): JSX.Element {
  if (pageProps?.statusCode) {
        return <Component {...pageProps} />
   }

  return (
    <Layout {...pageProps}>
      <Component {...pageProps} />
    </Layout>
  )
}

MyApp.getInitialProps = getInitialProps```

Reply:
OK, let's go like this, we will see with @MdechampG for the implementation of a custom error component